### PR TITLE
PSWF 2D Grid Construction

### LINKED ISSUE #74: Replace the grid generating method in PSWF basis
Current the PSWF basis uses a different method for generating 2d grid. Ideally it should be used the same one `grid_2d` as that of FB methods. But the internal representation of data has be mixed up by two statements in line 61 and 62, `x = y_2d_grid[points_in_disk]; y = x_2d_grid[points_in_disk]`. This exchange of x and y has consequence in implementing other functions in the PSWF class. Simply switching back and replacing with grid_2d do not work. Need to do more carefully work when optimizing the PSWF methods.

### PR DESCRIPTION
**Pull request by garrettwrong**
Converts manual grid to use our grid_2d and removes some redundant code.

Closes #74

### CODE REVIEW

**Comment by garrettwrong** [comment]
Considered additional work for this, but I think that isn't really related to just migrating the manual grids to use our utility.

Improving the prolate basis implementations can be considered after the minimal interoperation is achieved.

Opening to review.

File: src/aspire/basis/pswf_2d.py [hunk 1]
```diff
@@ -82,22 +82,11 @@ def _build(self):
...
-        x_2d_grid, y_2d_grid = np.meshgrid(x_1d_grid, x_1d_grid)
-        r_2d_grid = np.sqrt(np.square(x_2d_grid) + np.square(y_2d_grid))
-        points_in_disk = r_2d_grid <= self.rcut
-        x = y_2d_grid[points_in_disk]
-        y = x_2d_grid[points_in_disk]
-        self._r_disk = np.sqrt(np.square(x) + np.square(y)) / self.rcut
-        self._theta_disk = np.angle(x + 1j * y)
-        self._image_height = len(x_1d_grid)
-        self._disk_mask = points_in_disk
+        grid = grid_2d(self.nres, normalized=False, indexing="xy")
```
Comment (j-c-c) [inline src/aspire/basis/pswf_2d.py:86]: The xy-indexing is setting off alarm bells for me. I imagine this reproduces the original grid though. Should we be using yx everywhere?

**Review by j-c-c — COMMENTED** [review]
Looks good. Just one question about grid indexing.

File: src/aspire/basis/pswf_2d.py (hunk 1 above)
Comment (j-c-c) [reply src/aspire/basis/pswf_2d.py:86]: Maybe this is included in the "additional work" remark you made above.

Comment (garrettwrong) [reply src/aspire/basis/pswf_2d.py:86]: Screenshot 2023-10-30 at 12 40 36 PM

Comment (garrettwrong) [reply src/aspire/basis/pswf_2d.py:86]: hey, at least this got rid of lines like

```
        x = y_2d_grid[points_in_disk]
        y = x_2d_grid[points_in_disk]
```  

🤣

Comment (j-c-c) [reply src/aspire/basis/pswf_2d.py:86]: > hey, at least this got rid of lines like
> 
> ```
>         x = y_2d_grid[points_in_disk]
>         y = x_2d_grid[points_in_disk]
> ```
> 
> 🤣

An marked improvement for sure!

**Review by j-c-c — APPROVED** [review]

**Comment by garrettwrong** [comment]
Going to merge this in.  There is no functional change, the grids are identically created and dead code removed.

I have another branch that can update the xy to yx, but dont want to be forced to update the test references before the larger interop branch goes in. (This changes the coef indexing and reverses the rotations).

**Pull request merged**